src/state/slices/opportunitiesSlice/resolvers/thorchainsavers/index.ts

```diff
@@ -164,11 +171,71 @@ export const thorchainSaversStakingOpportunitiesMetadataResolver = async ({
   }
 
   if (getConfig().REACT_APP_FEATURE_RUNEPOOL) {
+    const { data: reservePositions } = await axios.get<ThorchainRunepoolReservePositionsResponse>(
+      `${getConfig().REACT_APP_MIDGARD_URL}/member/thor1dheycdevq39qlkxs2a6wuuzyn4aqxhve4qxtxt`,
+    )
     const { data: runepoolInformation } =
       await axios.get<ThorchainRunepoolInformationResponseSuccess>(
         `${getConfig().REACT_APP_THORCHAIN_NODE_URL}/lcd/thorchain/runepool`,
       )
 
+    const { data: poolsResponse } = await axios.get<ThornodePoolResponse[]>(
+      `${getConfig().REACT_APP_THORCHAIN_NODE_URL}/lcd/thorchain/pools`,
+    )
+
+    const poolsByAsset = poolsResponse.reduce<Record<string, ThornodePoolResponse>>(
+      (acc, pool) => ({ ...acc, [pool.asset]: pool }),
+      {},
+    )
+
+    const underlyingAssetIds = reservePositions.pools
+      .map(pool => poolAssetIdToAssetId(pool.pool))
+      .filter(
+        assetId => isSome(assetId) && opportunityIds.includes(assetId as OpportunityId),
+      ) as AssetId[]
+
+    const totalRuneAmount = reservePositions.pools.reduce((acc, pool) => {
+      const share = bnOrZero(pool.liquidityUnits).div(poolsByAsset[pool.pool].pool_units)
+      const runeAmount = share.times(poolsByAsset[pool.pool].balance_rune)
+
+      return acc.plus(runeAmount)
+    }, bn(0))
+
+    const { underlyingAssetRatiosBaseUnit, underlyingAssetWeightPercentageDecimal } =
+      reservePositions.pools.reduce(
+        (acc, pool, i) => {
```

Comment:
q: can we end up with wrong indexes here re: PoL-enabled assets here? i.e for safety, we should probably index this by THORChain pool/AssetId as well

Reply:
Absolutely! But it's not right anymore in [1274545](https://github.com/shapeshift/web/pull/7459/commits/1274545945102b423436a25ccdf5fcfc878b6644)